cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -325,3 +398,47 @@ def _handle_config_validation_error(e: ConfigValidationError) -> CreateClusterBa
             configuration_validation_errors=config_validation_messages, message="Invalid cluster configuration"
         )
     )
+
+
+def _analyze_changes(changes):
```

Comment:
shall we add a unit test?

Reply:
See above.

Reply:
Note: added `test_handling_of_change_sets` to check top level behavior against various changesets.